### PR TITLE
add conda-forge/label/jupyterlab_alpha for binder

### PR DESCRIPTION
**Pull request by bollwyvl**
[Binder](https://mybinder.org/v2/gh/bollwyvl/jupytercad/patch-1)

Cool package! Would love to try it out... and maybe get it working with [JupyterLite eventually](https://github.com/jupyterlite/jupyterlite/issues/826). This tries to get the binder building, though it was hanging last i checked...

### CODE REVIEW

**Comment by bollwyvl** [comment]
It looks like it would benefit from a `yarn.lock`-resolve and application of `yarn-deduplicate`...

**Comment by trungleduc** [comment]
Thanks @bollwyvl, i rebased to take the package resolution fix from `main`.

**Comment by bollwyvl** [comment]
Still seeing:

```
 > jupytercad@0.1.0 install:extension /home/jovyan
    > jlpm run build
    
    yarn run v1.22.19
    $ jlpm build:schema && jlpm run build:lib && jlpm run build:worker && jlpm run build:labextension:dev
    $ json2ts -i src/schema -o src/_interface --no-unknownAny --unreachableDefinitions --cwd ./src/schema && cd src/schema && node ../../schema.js
    $ tsc
    src/fcplugin/modelfactory.ts(28,5): error TS2322: Type '"FCStd"' is not assignable to type 'ContentType'.
    src/fcplugin/plugins.ts(61,14): error TS2322: Type 'Token<IJupyterCadTracker>' is not assignable to type 'Token<any>'.
      Types have separate declarations of a private property '_tokenStructuralPropertyT'.
    src/index.ts(23,3): error TS2322: Type 'import("/home/jovyan/node_modules/@lumino/coreutils/types/token").Token<import("/home/jovyan/src/token").IJupyterCadTracker>' is not assignable to type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/coreutils/types/token").Token<import("/home/jovyan/src/token").IJupyterCadTracker>'.
      Types have separate declarations of a private property '_tokenStructuralPropertyT'.
    src/index.ts(40,31): error TS2322: Type 'Token<IJupyterCadTracker>' is not assignable to type 'Token<any>'.
    src/jcadplugin/modelfactory.ts(28,5): error TS2322: Type '"jcad"' is not assignable to type 'ContentType'.
    src/jcadplugin/plugins.ts(57,14): error TS2322: Type 'Token<IJupyterCadTracker>' is not assignable to type 'Token<any>'.
    src/panelview/leftpanel.tsx(14,27): error TS2345: Argument of type 'ControlPanelHeader' is not assignable to parameter of type 'Widget'.
      The types of 'title.owner.layout' are incompatible between these types.
        Type 'import("/home/jovyan/node_modules/@lumino/widgets/types/layout").Layout | null' is not assignable to type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/layout").Layout | null'.
          Property '[Symbol.iterator]' is missing in type 'import("/home/jovyan/node_modules/@lumino/widgets/types/layout").Layout' but required in type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/layout").Layout'.
    src/panelview/objectproperties.tsx(13,11): error TS2345: Argument of type 'import("/home/jovyan/src/panelview/objectproperties").ObjectProperties.IOptions' is not assignable to parameter of type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/lib/components/panelwithtoolbar").PanelWithToolbar.IOptions'.
      Types of property 'layout' are incompatible.
        Type 'import("/home/jovyan/node_modules/@lumino/widgets/types/panellayout").PanelLayout | undefined' is not assignable to type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/panellayout").PanelLayout | undefined'.
          Type 'import("/home/jovyan/node_modules/@lumino/widgets/types/panellayout").PanelLayout' is not assignable to type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/panellayout").PanelLayout'.
    src/panelview/objecttree.tsx(58,11): error TS2345: Argument of type 'import("/home/jovyan/src/panelview/objecttree").ObjectTree.IOptions' is not assignable to parameter of type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/lib/components/panelwithtoolbar").PanelWithToolbar.IOptions'.
      Types of property 'layout' are incompatible.
        Type 'import("/home/jovyan/node_modules/@lumino/widgets/types/panellayout").PanelLayout | undefined' is not assignable to type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/panellayout").PanelLayout | undefined'.
          Property '[Symbol.iterator]' is missing in type 'import("/home/jovyan/node_modules/@lumino/widgets/types/panellayout").PanelLayout' but required in type 'import("/home/jovyan/node_modules/@jupyterlab/ui-components/node_modules/@lumino/widgets/types/panellayout").PanelLayout'.
    src/panelview/rightpanel.tsx(14,27): error TS2345: Argument of type 'ControlPanelHeader' is not assignable to parameter of type 'Widget'.
```

**Comment by bollwyvl** [comment]
yeah, i dunno... binder doesn't even start now

**Comment by trungleduc** [comment]
> yeah, i dunno... binder doesn't even start now

The issue seems to be related to `jupyter-server` 2.0 since binder still starts the `notebook` server. I can start the local container by using `repo2docker` with a custom `start` script. But the binder build is still not working.

File: binder/postBuild [hunk 1]
```diff
@@ -11,21 +11,33 @@
 import subprocess
 import sys
 from pathlib import Path
-
+import shutil
 
 ROOT = Path.cwd()
 
+
+# Hack to make binder work with jupyter-server 2
```
Comment (trungleduc) [inline binder/postBuild:19]: Finally i can make it work with this horrible hack :disappointed:

File: binder/postBuild [hunk 2]
```diff
@@ -11,21 +11,33 @@
...
+lab_script = shutil.which("jupyter-lab")
+notebook_script = shutil.which("jupyter-notebook")
+shutil.copy(lab_script, notebook_script)
+shutil.copy(ROOT / "binder" / "jupyter_config.json", ROOT)
+
+
```
Comment (trungleduc) [inline binder/postBuild:25]: Finally i can make it work with this horrible hack :disappointed:

**Comment by trungleduc** [comment]
@bollwyvl Binder works now, do you think we can go with the hack i used?

**Pull request merged**